zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -360,6 +360,12 @@ sealed trait Http[-R, +E, -A, +B] extends (A => ZIO[R, Option[E], B]) { self =>
   final def unwrap[R1 <: R, E1 >: E, C](implicit ev: B <:< ZIO[R1, E1, C]): Http[R1, E1, A, C] =
     self.flatMap(Http.fromZIO(_))
 
+  /**
+   * Applies Http based only if the condition function evaluates to true
+   */
+  final def when[A1 <: A](f: A1 => Boolean): Http[R, E, A1, B] =
+    Http.fromFunctionHExit[A1](a => if (f(a)) self.execute(a.asInstanceOf[A]) else HExit.empty)
```

Comment:
```suggestion
    Http.collectHttp[A1]({case a if f(a) => self else Http.empty)
```

If this doesn't give us desired performance, try creating a `When` primitive.

Reply:
Using `fromFunctionExit`:
```
[info] Benchmark                                Mode  Cnt        Score        Error  Units
[info] HttpRouteTextPerf.benchmarkHttpProgram  thrpt    3  7397922.838 ? 575478.018  ops/s
```

`collectHttp` is not giving desired performance
```
[info] Benchmark                                Mode  Cnt        Score        Error  Units
[info] HttpRouteTextPerf.benchmarkHttpProgram  thrpt    3  3018260.987 ? 387426.234  ops/s
```